### ISSUE TITLE
Go to definition not working

### ISSUE THREAD

**Issue by svenbs** [lingua-pupuli/puppet-vscode#331]
### What Versions are you running?

OS Version: Ubuntu 17.10
VSCode Version: Version: 1.25.1 Commit: 1dfc5e557209371715f655691b1235b6b26a06be
Puppet Extension Version: v1.11.0
Puppet Agent Version: 5.5.3
PDK Version (if applicable): 1.6.0

### What You Are Seeing?

I want to use the "Go to Definition" or "Peek Definition" Feature.
When I try to go to a definition of a module it says: "No definition found for 'module'"

### What is Expected?

I expect it to open the module.

### How Did You Get This To Happen? (Steps to Reproduce)

Open a .pp file inside your repo. Click an include statement and press F12 (or your binding for "Go to Definition".

### Output Log

DEBUG: 2018-07-26T15:01:31.078Z Found a valid Puppet installation at /opt/puppetlabs/puppet
DEBUG: 2018-07-26T15:01:31.078Z Configuring commands
DEBUG: 2018-07-26T15:01:31.079Z [createLanguageServerProcess] Language server found at: /home/sbuesing/.vscode/extensions/jpogran.puppet-vscode-0.11.0/vendor/languageserver/puppet-languageserver
DEBUG: 2018-07-26T15:01:31.080Z [getRubyExecFromPuppetAgent] Using environment variable RUBY_DIR=/opt/puppetlabs/lib/ruby
DEBUG: 2018-07-26T15:01:31.080Z [getRubyExecFromPuppetAgent] Using environment variable PATH=/opt/puppetlabs/puppet/bin:/opt/puppetlabs/facter/bin:/opt/puppetlabs/bin:/opt/puppetlabs/lib/ruby/bin:/opt/puppetlabs/sys/tools/bin:/usr/local/go/bin:/home/sbuesing/gopkgs/bin:/home/sbuesing/bin:/home/sbuesing/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/opt/puppetlabs/bin
DEBUG: 2018-07-26T15:01:31.080Z [getRubyExecFromPuppetAgent] Using environment variable RUBYLIB=/opt/puppetlabs/puppet/lib:/opt/puppetlabs/facter/lib:
DEBUG: 2018-07-26T15:01:31.080Z [getRubyExecFromPuppetAgent] Using environment variable RUBYOPT=rubygems
DEBUG: 2018-07-26T15:01:31.080Z [getRubyExecFromPuppetAgent] Using environment variable SSL_CERT_FILE=/opt/puppetlabs/puppet/ssl/cert.pem
DEBUG: 2018-07-26T15:01:31.080Z [getRubyExecFromPuppetAgent] Using environment variable SSL_CERT_DIR=/opt/puppetlabs/puppet/ssl/certs
DEBUG: 2018-07-26T15:01:31.080Z [createLanguageServerProcess] STDIO Server process starting
DEBUG: 2018-07-26T15:01:31.080Z [startLanguageServerProcess] Starting the language server with ruby /home/sbuesing/.vscode/extensions/jpogran.puppet-vscode-0.11.0/vendor/languageserver/puppet-languageserver --stdio --timeout=10 --local-workspace=/home/sbuesing/git/department/puppet-control --enable-file-cache
DEBUG: 2018-07-26T15:01:31.086Z [startLanguageServerProcess] Language server PID:24429
DEBUG: 2018-07-26T15:01:31.086Z LanguageServer Process Started: 24429
DEBUG: 2018-07-26T15:01:31.087Z Starting STDIO client: 
DEBUG: 2018-07-26T15:01:31.087Z Configuring language server options
DEBUG: 2018-07-26T15:01:31.087Z Starting language server client (stdio)
DEBUG: 2018-07-26T15:01:31.087Z Configuring language server client options
DEBUG: 2018-07-26T15:01:31.092Z [Puppet Lang Server Client] stdio connected
DEBUG: 2018-07-26T15:01:31.092Z Congratulations, your extension "vscode-puppet" is now active!
DEBUG: 2018-07-26T15:01:31.284Z pdk shell started, pid: 24441
DEBUG: 2018-07-26T15:01:31.818Z Language server client started, setting puppet version

**Comment by michaeltlombardi** [comment]
Can you clarify what resource you're trying to find the definition of? It _seems_ from the context like you are right clicking on the word `module` in your manifest?

If I right-click on `file` in the example below I can peek or go to the definition:

```puppet
file {'/home/someone/file':
  ensure => present,
}
```

I believe peek/go-to only works for specific resources, not overall modules. Does the example here work for you or is the functionality entirely broken on your end?

**Comment by svenbs** [comment]
@michaeltlombardi : The functionality is completely broken. I cant peek or go to definition for standard puppet-resources, as you described, either.
So clicking on file "peek definition" gives the same message as stated above.

**Comment by wiccan2** [comment]
I am having the same issue, can't go to definition of any resources at all.

**Comment by wiccan2** [comment]
@jpogran Yes this is still an issue on 0.12.

**Comment by jpogran** [comment]
@wiccan2 I apologize, I got confused when replying. Do you have the same platform and versions as the original reporter?

@svenbs Apologies for the confusion, is this still and issue for you? We've recently finished the 0.13 release and I'm adding this to the pile to be investigated now.

**Comment by jpogran** [comment]
So, I spun up an ubuntu 18.04, vscode 1.28, and pdk 1.7.1 and reproduced the issue. @glennsarti I think we have a problem with the file uri being sent, it seems to have one too many slashes after `file:` I typed a simple `file` declaration and watched the log and I see the following:

```
D, [2018-10-24T16:11:25.404160 #34999] DEBUG -- : --- INBOUND
{"jsonrpc":"2.0","id":9,"method":"textDocument/definition","params":{"textDocument":{"uri":"file:///home/james/Documents/foo.pp"},"position":{"line":0,"character":2}}}
---
D, [2018-10-24T16:11:25.418013 #34999] DEBUG -- : --- OUTBOUND
{"jsonrpc":"2.0","id":9,"result":[{"uri":"file:////opt/puppetlabs/pdk/share/cache/ruby/2.4.0/gems/puppet-6.0.2/lib/puppet/type/file.rb","range":{"start":{"line":14,"character":0},"end":{"line":14,"character":1024}}}]}
---
D, [2018-10-24T16:11:25.554092 #34999] DEBUG -- : --- INBOUND
{"jsonrpc":"2.0","id":10,"method":"textDocument/hover","params":{"textDocument":{"uri":"file:///home/james/Documents/foo.pp"},"position":{"line":0,"character":5}}}
---
```

Note the file uri differences between the definition request and the hover request.

I checked the path, and the file is present at that location, so it's not a pathing issue in detecting pdk folders.

I see on line https://github.com/lingua-pupuli/puppet-editor-services/blob/master/lib/puppet-languageserver/manifest/definition_provider.rb#L73 and line https://github.com/lingua-pupuli/puppet-editor-services/blob/master/lib/puppet-languageserver/manifest/definition_provider.rb#L88 we use `file:///` there. I think we may need to use a uri builder instead of making our own.

**Comment by jpogran** [comment]
@svenbs you are correct. :)

@wiccan2 @svenbs the fix is merged into master, it will be part of the (0.13.1](https://github.com/lingua-pupuli/puppet-vscode/milestone/17) release, which we should have out next week
